### DISCUSSION TITLE
Fetching inbox messages takes time

### DISCUSSION THREAD

**Discussion by tamili2i** [ecederstrand/exchangelib#1186] [Q&A] [answered]
While fetching inbox messages takes around 15 sec. Is there a way to reduce or optimize the response time?

I am using WorkMail + Exchangelib. 
Authentication - Using Basic Authentication.

**Comment by ecederstrand** [comment]
I would suggest to [enable debug logging](https://ecederstrand.github.io/exchangelib/#troubleshooting) so you can see exactly where time is spent in your script.

**Reply by tamili2i** [reply]
Thanks, I will check and let you know

**Reply by tamili2i** [reply]
[message.log](https://github.com/ecederstrand/exchangelib/files/11684248/message.log)
Hi @ecederstrand , Here is my debug log it took around 3mins for fetching 100 mails. I am not able to find where the delay is happening. But i see some logs like
`
DEBUG:exchangelib.util:Session 22225 thread 4693194240: retry 0 timeout 120 POST'ing to https://ews.mail.us-west-2.awsapps.com/EWS/Exchange.asmx after 10s wait

DEBUG:exchangelib.util:Retry: 0
Waited: 10
Timeout: 120
Session: 22225
Thread: 4693194240`
I am not sure what's happing here. I don't know how to overcome this issue.
Please let me know.

**Reply by ecederstrand** [reply]
Try updating exchangelib. You seem to be on an old version.

The logs will tell you whether there are HTTP requests timing out or being very slow. If so, you could try using e.g. `.only()` to limit the amount of data fetched. Otherwise it's just a slow server witch is difficult to do anything about.

**Reply by tamili2i** [reply]
Hi @ecederstrand , i have updated to latest exchangelib. here is the log after that. it took 8sec to fetch 4 messages.
[messages_1.log](https://github.com/ecederstrand/exchangelib/files/11702553/messages_1.log)

**Reply by tamili2i** [reply]
Any update on this @ecederstrand ?

**Reply by ecederstrand** [reply]
I don't think 8 seconds for a full auth + fetch of data sounds unreasonable. If you have specific actions in the log that you think are slow and are not just caused by slow responses from the server which exchangelib cannot do anything about, and you have implemented the `.only()` performance improvement I suggested, then let's look at the specifics of that slowness.

**Reply by tamili2i** [reply]
Yeah, I have implemented .only(). I am not sure where the delay is. so i shared the debug log.

**Reply by ecederstrand** [reply, marked as answer]
In the logs, I only see a server taking some seconds to respond to each request. Not something related to exchangelib-specific slowness.

**Reply by tamili2i** [reply]
Got it. Then it has to be something related to a mail server that is Workmail for us.
